yetanotherco/aligned_layer
# docs: validate public input first version

# Description

- This PR adds an example of validating public input of a Risc0 proof in a smart contract.

# To Test

- Follow the examples guide in the docs. Remember, to send the required arguments to the `verifyBatchInclusion` method in the contract, you first need to encode them to hex.
To do so, you can use the following nodejs function:
```javascript
function bytesToHex(bytes) {
    let hex = [];
    for (let i = 0; i < bytes.length; i++) {
        let current = bytes[i] < 0 ? bytes[i] + 256 : bytes[i];
        hex.push((current >>> 4).toString(16));
        hex.push((current & 0xF).toString(16));
    }
    return hex.join("");
}
```
You will also need to encode the `pubInputBytes` to hex. This is the `risc_zero_fibonacci.pub` file generated with the proof. For the proof we are generating, it is `0xf404000086070000` so you can use that value.
-Then, use cast to interact with the contract:
```
cast send --rpc-url https://ethereum-holesky-rpc.publicnode.com <CONTRACT_ADDRESS> "verifyBatchInclusion(bytes32,bytes32,bytes32,bytes20,bytes32,bytes,uint256,bytes)" <proofCommitment> <pubInputCommitment> <programIdCommitment> <proofGeneratorAddr> <batchMerkleRoot> <merkleProof> <verificationDataBatchIndex> <pubInputBytes> --private-key <PRIVATE_KEY>
```
- This transaction should take a little time, after that, you can check the contract address in Etherscan Holesky and you will see the event emitted with the two hex values corresponding to the two last numbers of the Fibonacci sequence.
